src/main/webapp/search-meal-script.js

```diff
@@ -13,9 +13,14 @@
 // limitations under the License.
 
 function searchMeal() {
-  fetch("/meal")
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const searchRequest = urlParams.get("query") ?? "";
+  fetch(`/meal/query=${searchRequest}`)
     .then((response) => response.json())
     .then((dishes) => {
+      const amount = document.getElementById('amount-block');
+      amount.innerText = `${dishes.length} results`;
```

Comment:
The word "results" should become "result" when there's exactly 1 result.

Reply:
The simple way to do it is to just use the ternary operator:
```
const count = dishes.length;
const isSingular = count == 1;
const text = `${dishes.length} result${isSingular? "" : "s"}`;
```

The cleaner approach would be to extract it to a function.

The most future-proof solution would be to use an external library: https://github.com/plurals/pluralize
You can load the script with a regular `script src` tag, using this url as the path: https://raw.githubusercontent.com/plurals/pluralize/master/pluralize.js

However, I think either of the simpler approaches would do. There's no point introducing a library just to append an `"s"` once.